core/src/commonMain/kotlin/com/powersync/db/PowerSyncDatabaseImpl.kt

```diff
@@ -247,4 +254,38 @@ internal class PowerSyncDatabaseImpl(
             }
         }
     }
+
+    override suspend fun disconnect() {
+        if(syncJob != null && uploadJob != null && syncJob!!.isActive && uploadJob!!.isActive) {
+            //Wait for job to finish and then cancel
+            syncJob?.cancelAndJoin()
+            uploadJob?.cancelAndJoin()
+            syncStream = null
+            currentStatus.update(connected = false, connecting = false)
+        }
+    }
+
+    override suspend fun disconnectAndClear(clearLocal: Boolean) {
+        disconnect()
+
+        this.writeTransaction {
+            execute("DELETE FROM ${PsInternalTable.OPLOG}")
+            execute("DELETE FROM ${PsInternalTable.CRUD}")
+            execute("DELETE FROM ${PsInternalTable.BUCKETS}")
+            execute("DELETE FROM ${PsInternalTable.UNTYPED}")
+
+            val tableGlob = if (clearLocal) "ps_data_*" else "ps_data__*"
+            val existingTableRows = getAll(
```

Comment:
There is a very similar query executed here https://github.com/powersync-ja/powersync-kotlin/blob/main/core/src/commonMain/kotlin/com/powersync/bucket/BucketStorage.kt#L47

This query should be added to the `PsInternalDatabase` and called from there.

Reply:
I added the query to `PsInternalDatabase`, unsure if we want it to publicly accessible though.